src/main/java/de/tum/in/www1/artemis/web/rest/TextExerciseResource.java

```diff
@@ -189,6 +194,12 @@ public TextExerciseResource(TextExerciseRepository textExerciseRepository, TextE
             return createTextExercise(textExercise);
         }
 
+        // Validate score settings
+        Optional<ResponseEntity<TextExercise>> optionalScoreSettingsError = exerciseService.validateScoreSettings(textExercise);
```

Comment:
We are using this several times for the different exercise types. I think it would make sense to move this check completely to `ExerciseService.java`, e.g. as `checkScoreSettings()` and then you call this method instead of the 6 lines.